skiko/src/jvmMain/cpp/common/PathSegmentIterator.cc

```diff
@@ -19,32 +19,50 @@ extern "C" JNIEXPORT jlong JNICALL Java_org_jetbrains_skia_PathSegmentIteratorKt
     return static_cast<jlong>(reinterpret_cast<uintptr_t>(&deletePathSegmentIterator));
 }
 
-extern "C" JNIEXPORT jobject JNICALL Java_org_jetbrains_skia_PathSegmentIteratorKt_PathSegmentIterator_1nNext
-  (JNIEnv* env, jclass jclass, jlong ptr) {
+extern "C" JNIEXPORT void JNICALL Java_org_jetbrains_skia_PathSegmentIteratorKt_PathSegmentIterator_1nNext
+  (JNIEnv* env, jclass jclass, jlong ptr, jfloatArray result) {
     SkPath::Iter* instance = reinterpret_cast<SkPath::Iter*>(static_cast<uintptr_t>(ptr));
     SkPoint pts[4];
     SkPath::Verb verb = instance->next(pts);
-    jobject segment;
+
+    std::bitset<8> context(verb);
```

Comment:
What's that code with context? Why not return context as int from the method?

Reply:
this is basically an byte which encodes following values:
type of segment (first three bytes), flag for isCloseLine and flag for isClosed. 

we can, of course return it as a separate value but I though that this will compilcate signature for helper functions like (withResult) - I am not arguing - I have a lot of doubts about this kind of serialization/deseralization